ui/src/components/journey/index.tsx

```diff
@@ -0,0 +1,217 @@
+import { useState } from 'react';
+import { useMutation } from '@tanstack/react-query';
+import {
+  Box,
+  Typography,
+  Grid,
+  CardContent,
+  CardActions,
+  AlertColor,
+} from '@mui/material';
+import { DeleteRounded } from '@mui/icons-material';
+import { StyledButton, StyledGrid, StyledJourneyCard } from './styled';
+import { AlertPopUp, CustomButton } from '..';
+import ViewPassengerCard from './view_passender_card';
+import ViewDriverCard from './view_driver_card';
+import RouteHeadline from './route_headline';
+import ViewSearchCard from './view_search_card';
+import { IJourneyEntity, IJourneyProps } from '../../interfaces';
+import {
+  useAddPassenger,
+  useDeleteJourney,
+  useDeletePassenger,
+} from '../../hooks/useJourney';
+import { useUserProfile } from '../../hooks/usePlatformUser';
+import { httpStatus } from '../../constants';
+
+export default function Journey(props: IJourneyProps) {
+  const [journeys, setJourneys] = useState<IJourneyEntity[]>(props.journeys);
+  const [snackbarSeverity, setSnackbarSeverity] = useState<AlertColor>('info');
+  const [openSnackbar, setOpenSnackbar] = useState<boolean>(false);
+  const [snackbarMessage, setSnackbarMessage] = useState<string>('');
+  const { isSuccess: isUserProfileSuccess, data: isUserDataSuccess } =
+    useUserProfile();
+
+  const processSnackbar = (severity: AlertColor, message: string) => {
+    setSnackbarSeverity(severity);
+    setSnackbarMessage(message);
+    setOpenSnackbar(true);
+  };
+
+  const mutateAddPassenger = useMutation({
+    mutationFn: useAddPassenger,
+    onSuccess: (resp) => {
+      if (resp.status.toString() === httpStatus.OK) {
+        const journeyIdFromURL = findJourneyIdFromURL(resp.config.url);
+        if (journeyIdFromURL !== 0) {
+          setJourneys(
+            journeys &&
+              journeys.filter((journey) => journey.id !== journeyIdFromURL)
+          );
+          processSnackbar('info', 'Your journey was booked.');
+        }
+      }
+    },
+    onError: (error: Error) => {
+      if (error.message.endsWith(httpStatus.CONFLICT)) {
+        // TODO: display a different error msg when passenger is also the driver.
```

Comment:
is this comment still needed?

Reply:
Yes because I also need to change the server first to reflect in the UI which will take some time.